### PR TITLE
Torch classification

### PR DESCRIPTION
**Pull request by yinweisu**
GPU mem leak investigation needed

### CODE REVIEW

**Comment by yinweisu** [comment]
timm dependency is needed

**Comment by yinweisu** [comment]
GPU memory leak is related to the usage of Nvidia apex. Check out https://github.com/NVIDIA/apex/issues/439. Looks like this issue hasn't been resolved yet. By disabling apex, the memory usage becomes consistent

File: gluoncv/auto/data/dataset.py [hunk 1]
```diff
@@ -385,6 +397,42 @@ def __getitem__(self, idx):
...
+        im_path = self._dataset['image'][idx]
+        img = self._imread(im_path).convert('RGB')
+        label = None
+        # # pylint: disable=not-callable
+        if self.transform is not None:
+            img = self.transform(img)
+        if self._has_label:
+            label = self._dataset['label'][idx]
+        else:
+            label = torch.tensor(-1, dtype=torch.long)
```
Comment (zhreshold) [inline gluoncv/auto/data/dataset.py:433]: Is this -1 label used anywhere? What will timm do in such case?

**Review by zhreshold — COMMENTED** [review]
Generally looks good to me, I will need to run some local test myself and get back to u

File: gluoncv/auto/data/dataset.py (hunk 1 above)
Comment (yinweisu) [reply gluoncv/auto/data/dataset.py:433]: I'm just following timm's dataset implementation. The data loader requires label to be a tensor instead of None, and timm's dataset will give a -1 tensor in such case.
